tesseract_kinematics/include/tesseract_kinematics/core/utils.h

```diff
@@ -301,27 +302,30 @@ inline bool isWithinLimits(const Eigen::Ref<const Eigen::Matrix<FloatType, Eigen
  * @param limits The joint limits of the robot
  */
 template <typename FloatType>
-inline std::vector<FloatType> getRedundantSolutions(const FloatType* sol, const Eigen::MatrixX2d& limits)
+inline std::vector<Eigen::Matrix<FloatType, Eigen::Dynamic, 1>>
+getRedundantSolutions(const Eigen::Ref<const Eigen::Matrix<FloatType, Eigen::Dynamic, 1>>& sol,
+                      const Eigen::MatrixX2d& limits)
 {
-  auto dof = static_cast<int>(limits.rows());
   FloatType val;
-  std::vector<FloatType> redundant_sols;
-  for (int i = 0; i < dof; ++i)
+  std::vector<Eigen::Matrix<FloatType, Eigen::Dynamic, 1>> redundant_sols;
+  for (Eigen::Index i = 0; i < static_cast<Eigen::Index>(sol.size()); ++i)
   {
     val = sol[i];
     while ((val -= (2 * M_PI)) > limits(i, 0))
     {
-      std::vector<FloatType> new_sol(sol, sol + dof);
-      new_sol[static_cast<size_t>(i)] = val;
-      redundant_sols.insert(redundant_sols.end(), new_sol.begin(), new_sol.end());
+      Eigen::Matrix<FloatType, Eigen::Dynamic, 1> new_sol =
+          Eigen::Map<const Eigen::Matrix<FloatType, Eigen::Dynamic, 1>>(sol.data(), 6);
```

Comment:
Is it necessary to map an Eigen object to the same type of Eigen object? Why not just copy it?

```suggestion
      Eigen::Matrix<FloatType, Eigen::Dynamic, 1> new_sol = sol;
```

Also, it looks like you're hard-coding the DOF size to 6 when it seems like it should be the same length as `sol`

Reply:
Fixed.